subprojects/gradle/cd/src/gradleTest/kotlin/com/avito/ci/BaseCiStepsPluginTest.kt

```diff
@@ -0,0 +1,147 @@
+package com.avito.ci
+
+import com.avito.android.plugin.artifactory.artifactoryPasswordParameterName
+import com.avito.android.plugin.artifactory.artifactoryUserParameterName
+import com.avito.instrumentation.instrumentationPluginId
+import com.avito.test.gradle.TestProjectGenerator
+import com.avito.test.gradle.dependencies.GradleDependency
+import com.avito.test.gradle.module.AndroidAppModule
+import com.avito.test.gradle.module.AndroidLibModule
+import com.avito.test.gradle.plugin.plugins
+import java.io.File
+
+internal abstract class BaseCiStepsPluginTest {
+    protected lateinit var projectDir: File
+
+    @Suppress("MaxLineLength")
+    protected fun generateProjectWithConfiguredCiSteps(
+        uploadCrashlyticsMappingFileEnabled: Boolean = true
+    ) {
+        TestProjectGenerator(
+            plugins = plugins {
+                id("com.avito.android.impact")
+            },
+            modules = listOf(
+                AndroidAppModule(
+                    name = "appA",
+                    dependencies = setOf(GradleDependency.Safe.project(":shared")),
+                    plugins = plugins {
+                        classpathPlugin(
```

Comment:
i think this is misleading.
We mimic plugins block and now injects `buildscript` block inside plugins.
It looks wrong here.

I propose to add separate `buildscript` block to Module, or just use plugins block with additional settings.gradle code:

```
pluginManagement {
  resolutionStrategy {
        eachPlugin {
            val pluginId = requested.id.id
            when {
                pluginId == "com.google.gms.google-services" -> useModule("com.google.gms:google-services:4.3.3")
                pluginId == "com.google.firebase.crashlytics" -> useModule("com.google.firebase:firebase-crashlytics-gradle:2.7.1")
            }
        }
    }
}
```

Reply:
The current behavior is a black box: you want to add the plugin like a classpath dependency and you don't need to know how this is implemented under the hood. I will try to convince you to believe in it.

The decision to use `pluginManagement` is great. But with the current impl of TestProjectGenerator we have problems.
We add plugins to some `module` but pluginmanagement is in `settings.gradle`. 
If we will configure two places, the DSL will be harder to write and understand.

Separate buildscript block:
pros - like a Gradle 
cons - the implicit intention of why we add that classpath stuff

If I don't convince you I will prefer the buildscript variant

Reply:
We can rename our `plugins` block as a middle ground.

```
plugins = pluginConfiguration {
  withId("")
  applyWithBuildscript()
}
```
or something like that

to be more explicit that it's not a 1to1 plugins block definition

Reply:
I changed part with applyWithBuildscript but decided to do not change a `plugin` part